### ISSUE TITLE
Two hamburger buttons

### ISSUE THREAD

**Issue by 532910** [kubernetes/website#22021]
I believe two hamburger buttons confuses users on https://kubernetes.io/ru/docs/home

![image](https://user-images.githubusercontent.com/14617699/85473650-3a035f00-b5bc-11ea-811e-525c5772bc00.png)

**Comment by sftim** [comment]
Hi @532910 

My guess is that there are a few different approaches to address this.
What should solutions focus on?

**Comment by shuuji3** [comment]
My idea is replacing the second hamburger icon from the top with a caret icon (▼/▲).

This is an example of using `fa-caret-up` and `fa-caret-down` of Font Awesome (ref. [Triangle Icons | Font Awesome](https://fontawesome.com/icons?d=gallery&m=free&q=triangle)):

**initial collapsed state**
![fa-caret-down](https://user-images.githubusercontent.com/1425259/85507862-fd704b80-b62d-11ea-8813-0839f1532418.png)

**expanded state**
![fa-caret-up](https://user-images.githubusercontent.com/1425259/85507871-fe08e200-b62d-11ea-90d7-b3bc381bef6d.png)

**Why I replace the second one?**
Because it's more common there is a hamburger icon on the top right corner.

**Why I chose the caret (triangle) icon?**
When we tap the icon, the TOC area moves up and down with animation. Its move fits the triangle shape. I think the shape can imply the motion of up and down, so it's more intuitive.

**Comment by 532910** [comment]
First of all, the second hamburger button is related to the documentation and must not be grouped with the search field.
The search field is related to the whole site and must be above documentation section.
Ths second hamburger button is the TOC for documentation and must be placed the same line as `Documentation`
The triangle button should be nice for this.

![image](https://user-images.githubusercontent.com/14617699/85535076-89787800-b61a-11ea-880e-9e88d5ace935.png)

**Comment by shuuji3** [comment]
I like this modification. It sounds reasonable to me.

Putting a search box on the header allows users to search soon because it is always shown on the header. Currently, we cannot see the search box after scrolling the page, so we need to scroll the page to the top.

**Comment by sftim** [comment]
Consider basing the UI for the lower menu on [`<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) from HTML5.

(Maybe even _use_ `<details>`)

**Comment by 532910** [comment]
Still looks odd.

**Comment by shuuji3** [comment]
I tried to use `<details>` and `<summary>` elements. What do you think about the below design?

**Screenshot**

Before a tap:
<img src="https://user-images.githubusercontent.com/1425259/111133088-7731d600-85bd-11eb-9cdf-39519ef9e863.png" width="50%">

After a tap:
<img src="https://user-images.githubusercontent.com/1425259/111133097-79943000-85bd-11eb-83c6-22a0c8f96e5b.png" width="50%">
(intentionally elipsising several elements)

**Comment by sftim** [comment]
@shuuji3 did you have a pull request for this?

**Comment by sftim** [comment]
This is a great issue for anyone to work on. It'd be great to see progress here.

**Comment by shuuji3** [comment]
@sftim No, I have not prepared a PR for this issue. The above screenshot was a just experiment on the browser. I'm glad if someone can work on.

@kLamda do you have a time to work on this issue?

**Comment by kLamda** [comment]
Yeah, sure I have time time to work on this issue.

**Comment by shuuji3** [comment]
Thank you! 🙂

**Comment by shuuji3** [comment]
I tried to replicate the above "Table of Contents" screenshot again. Here is the difference of HTML structure:

**Before change**
<img width="1173" alt="current-html" src="https://user-images.githubusercontent.com/1425259/135993993-de004e33-5249-411a-8486-9fc2a5233c68.png">

**After change**
<img width="1160" alt="after-change" src="https://user-images.githubusercontent.com/1425259/135993980-fb953c33-bf30-45c4-bc96-12b57363d5f4.png">

I created a new `<details>` and `<summary>` elements and moved the `<nav id="td-section-nav">` element  to the place the next to the `<summary>` element inside the `<details>` like this:

```html
<details open="">
  <summary>Table of Contents</summary>
  <nav id="td-section-nav">...</nav>
</details>
```

In addition to this change, I think we need some modifications of classes and the second hamburger buttons etc.

I hope this would be helpful.

**Comment by sftim** [comment]
We ought to sort this.
/priority important-longterm
/remove-priority backlog

**Comment by Gauravpadam** [comment]
I'd like to give this a try, This hasn't been fixed yet. @sftim can this issue be reassigned due to inactivity?

**Comment by sftim** [comment]
> I'd like to give this a try, This hasn't been fixed yet. @sftim can this issue be reassigned due to inactivity?

For Kubernetes SIG Docs, **anyone** can work on **any issue** (our more jargon-y way of saying this is “no [cookie licking](https://devblogs.microsoft.com/oldnewthing/20091201-00/?p=15843)”) and open a pull request to propose a related change.

**Comment by sftim** [comment]
Also see: https://github.com/kubernetes/website/issues/41171

**Comment by ritorhymes** [comment]
I have a live solution ready on PR #54023 that closes this issue and addresses other significant UX frictions for mobile docs navigation as detailed in issue #53998.

It's a revamped mobile UX architecture, it doesn't add any new dependencies, and though it's a custom feature, it’s implemented via Hugo Pipes within the existing Docsy/Hugo asset pipeline, so it’s designed to remain compatible with ongoing Docsy-alignment and future Docsy upgrades.

I call it a “Bottom Bar”: a drawer-based UI that consolidates access to the TOC, search, and back-to-top into a fixed bar at the bottom of the viewport. Tapping the new “Table of Contents” button on this bar opens the TOC inside a bottom drawer (and tapping it again closes it).

*Notice the top-of-page search bar no longer has a hamburger icon for TOC control, and there is now a clearly labeled “Table of Contents” button in the center of the bottom bar with a list icon.*

<img width="800" alt="diagram of the bottom bar in the closed drawer state" src="https://github.com/user-attachments/assets/ea60daf0-73fd-4772-858a-e189ee578c58" />

<br /><br />

**Opened TOC drawer state**

<img width="400" alt="bottom bar table of contents drawer opened" src="https://github.com/user-attachments/assets/50a3f38b-9124-4fed-9da9-621c5f54906a" />

It provides users clarity over which button controls the TOC. The “Table of Contents” label is i18n-enabled and uses the existing locale system.

It also brings the navigational UX in closer parity with the desktop experience that always has the TOC and search accessible on screen via the sidebar; no more scrolling to the top of the page each time.

I have a live demo you're welcome to check it out on: [k8s.ritovision.com/docs](https://k8s.ritovision.com/docs)

**Comment by SayakMukhopadhyay** [comment]
I have opened a new discussion https://github.com/kubernetes/website/discussions/54635 which investigates a different direction, one which closely aligns with either Docsy or Bootstrap. As we are already trying to align closer with Docsy, creating a bottom nav is probably going to complicate the codebase instead of simplifying it.
